homeassistant/components/device_tracker/__init__.py

```diff
@@ -191,7 +191,11 @@ def see(self, mac=None, dev_id=None, host_name=None, location_name=None,
                 mac = mac.upper()
                 device = self.mac_to_dev.get(mac)
                 if not device:
-                    dev_id = util.slugify(host_name or '') or util.slugify(mac)
+                    if not dev_id:
```

Comment:
There is a reason we have this construct that you're removing, it's so we have a fallback which is now lost if a dev_id is passed in. Do this instead:
```python
dev_id = util.slugify(dev_id or host_name or '') or util.slugify(mac)
```